sys/net/network_layer/fib/fib.c

```diff
@@ -294,16 +299,19 @@ static int fib_signal_rrp(uint8_t *dst, size_t dst_size, uint32_t dst_flags)
     msg.content.ptr = (void *)&content;
 
     for (size_t i = 0; i < FIB_MAX_RRP; ++i) {
-        if (notify_rrp[i] != KERNEL_PID_UNDEF) {
+        if ((notify_rrp[i] != KERNEL_PID_UNDEF) && (prefix_rrp[i] != NULL)) {
             DEBUG("[fib_signal_rrp] send msg@: %p to pid[%d]: %d\n", \
                   msg.content.ptr, (int)i, (int)notify_rrp[i]);
 
-            /* the receiver, i.e. the RRP, MUST copy the content value.
-             * using the provided pointer after replying this message
-             * will lead to errors
-             */
-            msg_send_receive(&msg, &reply, notify_rrp[i]);
-            DEBUG("[fib_signal_rrp] got reply.");
+            /* do only signal a RRP if its registered prefix matches */
+            if (universal_address_compare(prefix_rrp[i], dst, &dst_size) == 0) {
```

Comment:
Shouldn't you use ``&prefix_size`` instead of ``&dst_size` to "chop off" the rest of ``dst`` outside the prefix?